src/p_user.c

```diff
@@ -492,14 +492,22 @@ void P_PlayerThink (player_t* player)
 
       // killough 2/8/98, 3/22/98 -- end of weapon selection changes
 
-      if (player->weaponowned[newweapon] && newweapon != player->readyweapon)
-
-	// Do not go to plasma or BFG in shareware,
-	//  even if cheated.
-
-	if ((newweapon != wp_plasma && newweapon != wp_bfg)
-	    || (gamemode != shareware) )
-	  player->pendingweapon = newweapon;
+        if (player->weaponowned[newweapon])
+        {
+            player->carouselweapon = newweapon;
```

Comment:
This eliminates the occasional “stuck” carousel. Really interested in your opinions on the ec88293 commit, what could go wrong?

Reply:
No, this approach is buggy.